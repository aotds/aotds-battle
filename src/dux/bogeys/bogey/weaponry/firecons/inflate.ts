import _ from 'lodash';

import { FireconsState, FireconsShorthand } from './types';

export default function inflate_firecons(shorthand: FireconsShorthand = 0): FireconsState {
    if (Array.isArray(shorthand)) return shorthand;

    return _.range(1, shorthand + 1).map(id => ({ id }));
}
