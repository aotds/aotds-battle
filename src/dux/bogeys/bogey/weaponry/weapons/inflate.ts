import { WeaponShorthand, WeaponState } from './types';

export default function inflate_weapons(shorthand: WeaponShorthand[] = []): WeaponState[] {
    let id = 1;
    return shorthand.map(s => ({ ...s, id: id++ })) as any;
}
