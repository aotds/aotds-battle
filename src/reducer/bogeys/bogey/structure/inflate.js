import u from 'updeep';
import fp from 'lodash/fp';
import _ from 'lodash';

const shields = u.if( fp.isArray,
    _.flow(
        u.map( (v,id) => u({id: id+1})(v) ),
        fp.keyBy('id'),
    )
);

export default function inflate(state) {
    let inflate_hull = x => typeof x === 'number' ? { current: x, max: x } : x;

    let i = 1;
    return u({ 
        hull: inflate_hull, 
        armor: inflate_hull,
        shields 
    })(state);
}
