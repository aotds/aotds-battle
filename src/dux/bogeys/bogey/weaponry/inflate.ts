import u from 'updeep';

import firecons from './firecons/inflate';

export default function inflate_weaponry(shorthand) {
    return u({ firecons }, shorthand);
}
