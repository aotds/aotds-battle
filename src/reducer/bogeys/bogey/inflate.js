import u from 'updeep';
import _ from 'lodash';
import fp from 'lodash/fp';

import structure from './structure/inflate';

const firecons = u.if( fp.isNumber,     
    fp.flow(
        fp.times(fp.add(1)),
        fp.keyBy(_.identity),
        u.map( id => ({id}) )
    )
);

const drive = u.if( fp.isNumber,     
    max => ({ max, current: max })    
);


export default u({ 
    drive,
    structure,
    weaponry: { firecons },
});
