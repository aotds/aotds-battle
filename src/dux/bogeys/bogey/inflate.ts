import u from 'updeep';

import weaponry from './weaponry/inflate';

export default function inflate_bogey(shorthand) {
    return u(
        {
            weaponry
        },
        shorthand,
    );
}
