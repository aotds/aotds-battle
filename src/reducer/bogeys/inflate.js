import u from 'updeep';
import fp from 'lodash/fp';
import _ from 'lodash';

import bogey from './bogey/inflate';

export default fp.flow([
    u.if( _.isArray, a => _.keyBy(a,'id') ),
    u.map(bogey),
]);
