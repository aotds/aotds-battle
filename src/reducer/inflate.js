import u from 'updeep';
import fp from 'lodash/fp';
import _ from 'lodash';

import bogeys from './bogeys/inflate';

export default u({
    bogeys
});
