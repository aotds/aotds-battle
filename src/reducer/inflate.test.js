import inflate from './inflate';

const debug = require('debug')('aotds');

test( 'inflate', () => {

    let inflated = inflate({
        game: {
        },
        bogeys: [
            { 
                id: 'enkidu',
                drive: 3,
                structure: { shields: [ 1, 1, 2 ] },
                weaponry: { firecons: 2 },
            }
        ]
    });

    expect(inflated).toMatchSnapshot();

});
