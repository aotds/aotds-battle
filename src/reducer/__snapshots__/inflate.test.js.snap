// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inflate 1`] = `
Object {
  "bogeys": Object {
    "enkidu": Object {
      "drive": Object {
        "current": 3,
        "max": 3,
      },
      "id": "enkidu",
      "structure": Object {
        "shields": Object {
          "1": Object {
            "id": 1,
          },
          "2": Object {
            "id": 2,
          },
          "3": Object {
            "id": 3,
          },
        },
      },
      "weaponry": Object {
        "firecons": Object {
          "1": Object {
            "id": 1,
          },
          "2": Object {
            "id": 2,
          },
        },
      },
    },
  },
  "game": Object {},
}
`;
